[#63798] File storages health status report  https://community.openpr… (#18871)
opf/openproject: docs/system-admin-guide/files/external-file-storages/health-status/README.md

### Connection validation for OneDrive/SharePoint 

> Every file storage for OneDrive/SharePoint has the ability to run connection checks. This test is triggered manually by clicking on **Run checks now** in the sidebar on the right side of the file storage's details view. This check is available after the file storage is fully configured.

Reviewer: > This check is available after the file storage is fully configured.

@mereghost @Kharonus I remember that one of the tests is literally testing, whether the storage is fully configured. Do we really only _offer_ running tests after the storage is completely configured?

Reviewer: I just tested this locally. I can also run checks for a not completely configured storage. I guess this sentence is here to say that this button is only available after going through the initial create dialog (which can be aborted prematurely easily).

My suggestion:

```suggestion
Every file storage for OneDrive/SharePoint has the ability to run connection checks. This test is triggered manually by clicking on **Run checks now** in the sidebar on the right side of the file storage's details view. This check is available from the UI to edit the storage.
```

(this change would have to be repeated for the nextcloud section as well)

![Recheck connection for OneDrive/SharePoint in OpenProject administration](openproject_file_storages_onedrive_run_checks_button.png)

### Health report for OneDrive/SharePoint

Once the check is finished, a full health report will be generated. Click **open full health report** to see the report in full detail and to download it. 

![Link to open full health report for OneDrive/SharePoint file storage in OpenProject administration](openproject_file_storages_onedrive_open_full_health_report.png)

Full health status report will give an overview of all checks that were performed grouped into three categories: 

- Basic configuration
- Authentication
- Automatically managed project folder configuration (if activated for this file storage)

> In the top left corner you can **Re-run all checks** or **download the health (text format)** report by clicking respective buttons. 

Reviewer: ```suggestion
In the top right corner you can **Re-run all checks** or **download the health (text format)** report by clicking respective buttons. 
```

### Connection validation for Nextcloud

> Same as OneDrive/SharePoint, every file storage for Nextcloud has the ability to run a connection test. This test is triggered manually by clicking on **Run checks now** in the sidebar on the right side of the file storage's details view. This check is available after the file storage is fully configured.

Author: ```suggestion
Same as OneDrive/SharePoint, every file storage for Nextcloud has the ability to run a connection test. This test is triggered manually by clicking on **Run checks now** in the sidebar on the right side of the file storage's details view. This check is available from the UI to edit the storage.
```

![Recheck connection for Nextcloud in OpenProject administration](openproject_file_storages_nextcloud_run_checks_link.png)

### Health report for Nextcloud

Once the check is finished, a full health report will be generated. Click **open full health report** to see the report in full detail and to download it. 

![Link to open full health report for Nexctloud file storage in OpenProject administration](openproject_file_storages_nextcloud_open_full_health_report_button.png)

Full health status report will give an overview of all checks that were performed grouped into three categories: 

- Basic configuration
- Authentication
- Automatically managed project folder configuration (if activated for this file storage)

> In the top left corner you can **Re-run all checks** or **download the health (text format)** report by clicking respective buttons. 

Reviewer: ```suggestion
In the top right corner you can **Re-run all checks** or **download the health (text format)** report by clicking respective buttons. 
```
